helix-core/src/main/java/org/apache/helix/task/AbstractTaskDispatcher.java

```diff
@@ -121,17 +120,6 @@ public void updatePreviousAssignedTasksStatus(
               instance, pId);
           continue;
         }
```

Comment:
Would it be a problem that this skipping check is now done after markPartitionCompleted/markPartitionError?

Reply:
This shouldn't be an issue because this check is specifically added before to bypass the scenario where we have two current states which non of the markPartitionCompleted/markPartitionError will not be hit because it is either Running or Dropped. Good question though.